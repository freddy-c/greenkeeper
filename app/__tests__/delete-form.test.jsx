import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import DeleteForm from "../delete-form";

jest.mock("react-dom", () => ({
    ...jest.requireActual("react-dom"),
    useFormState: () => [{}, ""],
    useFormStatus: () => ({ pending: false }),
}));

const formAction = jest.fn();

describe("DeleteForm", () => {
    const manufacturerId = "123e4567-e89b-12d3-a456-426614174000";

    test("renders correctly", () => {
        const tree = renderer
            .create(<DeleteForm manufacturerId={manufacturerId} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    // test("submits the form when the submit button is clicked", () => {
    //     const { getByText } = render(
    //         <DeleteForm manufacturerId={manufacturerId} />
    //     );
    //     const deleteButton = getByText("Delete");

    //     fireEvent.click(deleteButton);

    //     // Add assertions to check the form submission behavior
    // });
});
