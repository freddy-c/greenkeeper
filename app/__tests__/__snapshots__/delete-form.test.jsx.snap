// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteForm renders correctly 1`] = `
<form
  action=""
>
  <input
    name="id"
    type="hidden"
    value="123e4567-e89b-12d3-a456-426614174000"
  />
  <button
    aria-disabled={false}
    className="border hover:cursor-pointer"
    type="submit"
  >
    Delete
  </button>
</form>
`;
